File: SieveofAtkin.js
limit = parseInt(process.argv[2])
sieve = Array(limit + 1)
sqrt = parseInt(limit ** 0.5)
for (i = 1; i <= sqrt; i++) {
	isquare = i ** 2
	tsquare = isquare * 3
	for (j = 1; j <= sqrt; j++) {
		jsquare = j ** 2
		if ((n = 4 * isquare + jsquare) <= limit && [1, 5].includes(n % 12))
			sieve[n] = !sieve[n]
		if ((n = tsquare + jsquare) <= limit && n % 12 == 7) sieve[n] = !sieve[n]
		if (i > j && (n = tsquare - jsquare) <= limit && n % 12 == 11)
			sieve[n] = !sieve[n]
	}
}
for (i = 5; i < sqrt; i++) {
	if (sieve[i]) {
		isquare = i ** 2
		for (j = isquare; j <= limit; j += isquare) sieve[j] = false
	}
}
process.stdout.write("2, 3")
total = 2
for (i = 5; i <= limit; i++) {
	if (sieve[i]) {
		process.stdout.write(", " + i)
		total++
	}
}
console.log("\nTotal:", total)

File: Rust/SieveofAtkin.rs
fn main() {
	let limit = std::env::args().collect::<Vec<String>>()[1].parse::<u32>().unwrap();
	let mut sieve = vec![false; (limit + 1) as usize];
	let sqrt = ((limit as f32).sqrt()) as u32;
	for i in 1..=sqrt {
		let isquare = i * i;
		let triple_isquare = isquare * 3;
		for j in 1..=sqrt {
			let jsquare = j * j;
			let mut n = 4 * isquare + jsquare;
			if n <= limit && [1, 5].contains(&(n % 12)) {
				sieve[n as usize] = !sieve[n as usize]
			}
			n = triple_isquare + jsquare;
			if n <= limit && n % 12 == 7 {
				sieve[n as usize] = !sieve[n as usize]
			}
			n = triple_isquare - jsquare;
			if i > j && n <= limit && n % 12 == 11 {
				sieve[n as usize] = !sieve[n as usize]
			}
		}
	}
	for i in 5..sqrt {
		if sieve[i as usize] {
			let isquare = i * i;
			for j in (isquare..=limit).step_by(isquare as usize) {
				sieve[j as usize] = false
			}
		}
	}
	print!("2, 3");
	let mut total = 2u32;
	for p in (5..limit).filter(|x| sieve[*x as usize]) {
		print!(", {}", p);
		total += 1
	}
	println!("\nTotal: {}", total)
}
